code/script.js

```diff
@@ -261,75 +271,151 @@ const selectQuestion = () => {
   const category = questions.options[questions.selectedIndex].parentNode.label
   // This variable stores what option group (category) the question belongs to.
   // We also need a variable that stores the actual value of the question we've selected.
+  //console.log(category) //hair color, eye color, accessories, other
+   const value = questions.options[questions.selectedIndex].value
+  //console.log(value)// brown, blue, glasses, smoker
 
   if (category === 'hair color') {
     currentQuestion = {
       attribute: 'hairColor',
-      // value: ,
+      value: value,
       // 👆 add the value from the input here
       category: category,
     }
   } else if (category === 'eye color') {
     // Set this up your self
+    currentQuestion = {
+      attribute: 'eyeColor',
+      value: value,
+      // 👆 add the value from the input here
+      category: category,
+    }
   } else if (category === 'accessories') {
     currentQuestion = {
-      //attribute: ,
+      attribute: value,
       // 👆 this is the property of the booleans such as smoke, glasses and hat. add the value from the input here
       value: true, // we're asking if this person wears a hat for exaple, so always true in the question.
       category: category,
     }
   } else if (category === 'other') {
     // Set this up your self (should be same structure as above)
+    currentQuestion = {
+      attribute: value,
+      value: true,
+      category: category,
+    }
   }
 }
 
 // This function should be invoked when you click on 'Find Out'.
 const checkQuestion = () => {
-  // Compare the currentQuestion with the secret person.
-  // See if we should keep or remove people based on that
-  // Then invoke filterCharacters
+  //console.log(secret)
+  //console.log(secret[currentQuestion.attribute]) //prints out the attribute of secret person (eg. brown hair, glasses: true, brown eyes)
+  
+  // *Compare the currentQuestion with the secret person.
+  //console.log(currentQuestion.value) // prints out the value of the current question (eg. purple hair, glasses: false, blue eyes)
+  // *See if we should keep or remove people based on that
+  //* Then invoke filterCharacters
+  if (secret[currentQuestion.attribute] === currentQuestion.value) { //if they are the same 
+    filterCharacters(true)
+  } else {
+    filterCharacters(false)
+  }  
```

Comment:
nice! 👍 I should take a peek at this and remember it for coming projects

Reply:
`filterCharacters(secret[currentQuestion.attribute] === currentQuestion.value)` would save you from bouilding a whole `if/else` block 💨